log/packetlogger.go

```diff
@@ -0,0 +1,157 @@
+package log
+
+import (
+	"os"
+
+	"github.com/go-logr/logr"
+	"github.com/go-logr/zapr"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+// WithLogLevel sets the log level
+func WithLogLevel(level string) LoggerOption {
+	return func(args *PacketLogger) { args.LogLevel = level }
+}
+
+// WithOutputPaths adds output paths
+func WithOutputPaths(paths []string) LoggerOption {
+	return func(args *PacketLogger) { args.OutputPaths = paths }
+}
+
+// WithServiceName adds a service name a logged field
+func WithServiceName(name string) LoggerOption {
+	return func(args *PacketLogger) { args.ServiceName = name }
+}
+
+// WithKeysAndValues adds extra key/value fields
+func WithKeysAndValues(kvs []interface{}) LoggerOption {
+	return func(args *PacketLogger) { args.KeysAndValues = append(args.KeysAndValues, kvs...) }
+}
+
+// WithEnableErrLogsToStderr sends .Error logs to stderr
+func WithEnableErrLogsToStderr(enable bool) LoggerOption {
+	return func(args *PacketLogger) { args.EnableErrLogsToStderr = enable }
+}
+
+// WithEnableRollbar sends error logs to Rollbar service
+func WithEnableRollbar(enable bool) LoggerOption {
+	return func(args *PacketLogger) { args.EnableRollbar = enable }
+}
+
+// WithRollbarConfig customizes the Rollbar details
+func WithRollbarConfig(config RollbarConfig) LoggerOption {
+	return func(args *PacketLogger) { args.RollbarConfig = config }
+}
+
+// PacketLogger is a wrapper around zap.SugaredLogger
+type PacketLogger struct {
+	logr.Logger
+	LogLevel              string
+	OutputPaths           []string
+	ServiceName           string
+	KeysAndValues         []interface{}
+	EnableErrLogsToStderr bool
+	EnableRollbar         bool
+	RollbarConfig         RollbarConfig
```

Comment:
shouldn't these be unexported? Its not like they can actually be set/modified and take effect since the config is done in `NewPacketLogger`

Reply:
ah yes, good catch. will update